packages/dx-grid-core/src/plugins/virtual-table/helpers.ts

```diff
@@ -39,11 +39,17 @@ export const getViewport: GetViewportFn = (
   // NOTE: prevent unnecessary updates
   // e.g. when rows changed but bounds remain the same.
   let result = viewport;
-  if (viewportTop !== viewport.viewportTop) {
-    result = { ...result, viewportTop };
+  if (viewportTop !== viewport.top) {
+    result = { ...result, top: viewportTop };
   }
-  if (viewportLeft !== viewport.viewportLeft) {
-    result = { ...result, viewportLeft };
+  if (viewportLeft !== viewport.left) {
+    result = { ...result, left: viewportLeft };
+  }
+  if (containerWidth !== viewport.width) {
+    result = { ...result, width: containerWidth };
+  }
+  if (containerHeight !== viewport.height) {
+    result = { ...result, height: containerHeight };
```

Comment:
Can we use the `return` on each `if` condition?

Reply:
No, we assemble viewport by pieces

Reply:
This is a horror 😱

Reply:
The way to avoid rerenders